components/table/src/table/table.component.ts

```diff
@@ -118,7 +118,7 @@ const NZ_CONFIG_COMPONENT_NAME = 'table';
     '[class.ant-table-wrapper]': 'true'
   }
 })
-export class NzTableComponent implements OnInit, OnDestroy, OnChanges, AfterViewInit {
+export class NzTableComponent<T = NzSafeAny> implements OnInit, OnDestroy, OnChanges, AfterViewInit {
```

Comment:
NzTableData would give better snippets when using a.b.c.d in the template

Reply:
The ng-language-service can not recognize generics. but this PR can make the project type-safe at compile(ivy) time.

https://github.com/angular/vscode-ng-language-service/issues/737